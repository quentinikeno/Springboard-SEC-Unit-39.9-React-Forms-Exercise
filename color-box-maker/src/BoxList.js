import React, { useState } from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";

const BoxList = () => {
	const [boxes, setBoxes] = useState([
		{ width: "100px", height: "100px", color: "green" },
	]);
	const addBox = (newBox) => {
		setBoxes((boxes) => [...boxes, newBox]);
	};
	return (
		<div>
			<h1>Color Box Maker</h1>
			<div>
				{boxes.map((box) => (
					<Box
						width={box.width}
						height={box.height}
						backgroundColor={box.color}
					/>
				))}
			</div>
			<NewBoxForm addBox={addBox} />
		</div>
	);
};

export default BoxList;
