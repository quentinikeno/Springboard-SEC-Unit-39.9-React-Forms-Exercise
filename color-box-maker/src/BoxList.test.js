import { render } from "@testing-library/react";
import BoxList from "./BoxList";

test("rendering without crashing", () => {
	render(<BoxList />);
});

test("matching the snapshot", () => {
	const { asFragment } = render(<BoxList />);
	expect(asFragment()).toMatchSnapshot();
});
