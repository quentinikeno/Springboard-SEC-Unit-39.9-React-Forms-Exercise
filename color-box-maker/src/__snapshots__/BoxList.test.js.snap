// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Color Box Maker
    </h1>
    <form>
      <div>
        <label
          for="width"
        >
          Width
        </label>
        <input
          id="width"
          max="500"
          min="1"
          name="width"
          type="range"
          value="250"
        />
        <span>
          250 pixels
        </span>
      </div>
      <div>
        <label
          for="height"
        >
          Height
        </label>
        <input
          id="height"
          max="500"
          min="1"
          name="height"
          type="range"
          value="250"
        />
        <span>
          250 pixels
        </span>
      </div>
      <div>
        <label
          for="backgroundColor"
        >
          Background Color
        </label>
        <input
          id="backgroundColor"
          name="backgroundColor"
          placeholder="Background Color"
          type="text"
          value=""
        />
      </div>
      <button>
        Create a new box!
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
